### PR TITLE
prov/psm3: Update the psm3 provider to match SRPM in IEFS 11.3 release

### PR DESCRIPTION
**Pull request by acgoldma**
Remove untested/internal code (PSM_OPA)
Fix VERSION file

Signed-off-by: Adam Goldman <adam.goldman@intel.com>

### CODE REVIEW

File: prov/psm3/psm3/hal_verbs/verbs_ep.c [hunk 1]
```diff
@@ -2000,23 +2000,6 @@ static psm2_error_t verbs_open_dev(psm2_ep_t ep, int unit, int port, int addr_in
 	}
 
 #if defined(USE_RC)
```
Comment (j-xiong) [inline prov/psm3/psm3/hal_verbs/verbs_ep.c:2002]: This outer #if can also be removed.

File: prov/psm3/psm3/hal_verbs/verbs_ep.h [hunk 2]
```diff
@@ -300,10 +300,6 @@ struct psm3_verbs_ep {
 	//uint8_t link_layer;         // IBV_LINK_LAYER_ETHERNET or other
 	uint8_t active_rate;
 #if defined(USE_RC)
```
Comment (j-xiong) [inline prov/psm3/psm3/hal_verbs/verbs_ep.h:302]: This outer #if can also be removed.

**Review by j-xiong — COMMENTED** [review]
Glad to see the removal of OPA related code. I would expect that be more clearly expressed in the  commit message.

File: prov/psm3/psm3/hal_verbs/verbs_ep.h (hunk 2 above)
Comment (acgoldma) [reply prov/psm3/psm3/hal_verbs/verbs_ep.h]: done

File: prov/psm3/psm3/hal_verbs/verbs_ep.c (hunk 1 above)
Comment (acgoldma) [reply prov/psm3/psm3/hal_verbs/verbs_ep.c]: done

**Comment by acgoldma** [comment]
> Glad to see the removal of OPA related code. I would expect that be more clearly expressed in the commit message.

OPA Code is ifdef by PSM_OPA define, we normally remove that before upstreaming.

**Review by j-xiong — APPROVED** [review]
Intel CI is having issue now. Wait till that is fixed.

**Pull request merged**